ixmp/tests/core/test_scenario.py

```diff
@@ -171,11 +171,26 @@ def test_scalar(self, scen):
         assert scen.scalar("f") == {"unit": "USD/km", "value": 90}
 
     # Store data
-    def test_add_par(self, scen):
-        # add_par() broadcasts scalar values/units across multiple keys
+    @pytest.mark.parametrize(
+        "args, kwargs",
+        (
+            # Scalar values/units are broadcast across multiple keys
+            (("b", ["new-york", "chicago"]), dict(value=100, unit="cases")),
+            # Empty DataFrame can be added without error
+            (("b", pd.DataFrame(columns=["i", "j", "value", "unit"])), dict()),
+        ),
+    )
+    def test_add_par(self, scen, args, kwargs):
+        scen = scen.clone(keep_solution=False)
+        scen.check_out()
+        scen.add_par(*args, **kwargs)
+
+    def test_add_par2(self, scen):
         scen = scen.clone(keep_solution=False)
         scen.check_out()
-        scen.add_par("b", ["new-york", "chicago"], value=100, unit="cases")
+        # Create a parameter with duplicate indices
+        scen.init_par("foo", idx_sets=["i", "i", "j"], idx_names=["i0", "i1", "j"])
```

Comment:
FYI @behnam-zakeri this new test gives me: `ValueError: Wrong number of items passed 3, placement implies 1`—same as you reported at https://github.com/iiasa/ixmp/issues/373#issuecomment-694195877.

So the error specifically occurs when the (a) the input data frame empty and (b) the parameter has ≥2 dimensions indexed by the same set, but with different names. Without (b) it doesn't occur.

With this pinned down, I can now look for a fix.